### PR TITLE
Do not cache registered namespace in local .azure\azureprofile.json

### PR DESCRIPTION
**Pull request by yugangw-msft**
Rather detect the error of 409 and give out an instruction which tell users to execute provider register command. Caching is generally disliked, because of not simple coding logic, command parallel running scenario support, etc.
Also modify"provider [register/unregister]" commands to polling the provider state to ensure the command went through

### CODE REVIEW

**Comment by yugangw-msft** [comment]
I tried and decided not to write an extra test for verifying the error message. The reason is that "unregistering a provider" could take a very long time, and the command might exit with a warning the provider still staying at "unregistering", when the test continues to create a resource, it will land up with a different  error. I can hack it up to get the test going, but it will sure cause confusion when someone else tries to run the test live sometime later. Unregistering a provider is not a common scenario anyway.

File: lib/util/profile/subscription.js [hunk 1]
```diff
@@ -263,15 +217,33 @@ _.extend(Subscription.prototype, {
...
-        });
+      self._waitForRegistrationComplete(client, false, numRetries, namespace, function (err) {
+        if (err) {
+          log.warn(util.format($('Unregistration is still ongoing. You can execute \'Azure Provider Show\' to monitor the status')));
+        }
+        callback();
+      });
+    });
+  },
+
```
Comment (amarzavery) [inline lib/util/profile/subscription.js]: Please document the function.
```
/**
 *
 *
*/
```

File: lib/util/profile/subscription.js [hunk 2]
```diff
@@ -263,15 +217,33 @@ _.extend(Subscription.prototype, {
...
       }
-      //remove the namespace from the list of registered resource namespaces
-      if (self.registeredResourceNamespaces) {
-        self.registeredResourceNamespaces = 
-          self.registeredResourceNamespaces.filter(function (item) {
-            return (!utils.ignoreCaseEquals(item, namespace));
-        });
+      self._waitForRegistrationComplete(client, false, numRetries, namespace, function (err) {
+        if (err) {
+          log.warn(util.format($('Unregistration is still ongoing. You can execute \'Azure Provider Show\' to monitor the status')));
```
Comment (amarzavery) [inline lib/util/profile/subscription.js]: azure --> 'a' should be lower case

File: lib/util/utils.js [hunk 3]
```diff
@@ -211,6 +212,19 @@ function createFollowRedirectFilter() {
 
 exports.createFollowRedirectFilter = createFollowRedirectFilter;
 
+polishErrorCausedByArmProviderNotRegistered = function () {
+  return function handle(resource, next, callback) {
+    var nextStream = next(resource, function (err, response, body) {
+      if (err && response && response.statusCode === 409) {
+        err.message = util.format('%s You can register it by executing the command of \'azure provider register\'.', 
+                                  err.message);
```
Comment (amarzavery) [inline lib/util/utils.js]: fix indentation

Comment (yugangw-msft) [inline lib/util/utils.js]: sorry, i saw it wrong. The indent is correct as it is the last function  parameter

File: lib/util/profile/subscription.js (hunk 2 above)
Comment (yugangw-msft) [inline lib/util/profile/subscription.js]: fixed

File: lib/util/profile/subscription.js (hunk 1 above)
Comment (yugangw-msft) [inline lib/util/profile/subscription.js]: added

**Pull request merged**